src/theme/ds-theme-provider.tsx

```diff
@@ -0,0 +1,68 @@
+import { ReactNode, useEffect, useState } from "react";
+import { ThemeProvider } from "styled-components";
+import { ColourMode, ThemeSpec } from "./types";
+import { getSystemColourMode } from "./use-theme-mode";
+
+export interface DSThemeProviderProps {
+    theme: ThemeSpec;
+    children: ReactNode;
+}
```

Comment:
could this be moved to `types.ts`? it's a convention to put public types there, and would help avoid accidental edits